libopae/plugins/xfpga/feature_plugins/feature_api.c

```diff
@@ -133,18 +133,22 @@ xfpga_fpgaFeatureEnumerate(fpga_handle handle, fpga_feature_properties *prop,
 	struct _fpga_feature_token *_ftoken;
 	struct DFH dfh;
 	int errors;
-	
+
 	ASSERT_NOT_NULL(prop);
-	ASSERT_NOT_NULL(tokens);
-	ASSERT_NOT_NULL(num_matches);	
-  
+	ASSERT_NOT_NULL(num_matches);
+
+	if ((max_tokens > 0) && (NULL == tokens)) {
+		FPGA_MSG("max_tokens > 0 with NULL tokens");
+		return FPGA_INVALID_PARAM;
+	}
+
 	*num_matches = 0;
-	
+
 	errors = feature_plugin_mgr_initialize(handle);
 	if (errors) {
 		FPGA_ERR("Feature token initialize errors");
 		result = FPGA_EXCEPTION;
```

Comment:
Even when setting result to FPGA_EXCEPTION here, it can be overwritten below, eg line 154.

Reply:
Correct. Should return at errors.